### PR TITLE
feat: add `useCreateComment` 

### PR DESCRIPTION
**Pull request by desfero**
- Adds `useCreateComment` hook inspired by the way `useCreatePost` works
- Adds follow UI example in `web-wagmi`
- Adds new docs for review https://dash.readme.com/project/lensprotocol/v1.1/docs/use-create-comment

### CODE REVIEW

File: examples/web-wagmi/src/components/Breadcrumbs.tsx
Comment (desfero) [inline examples/web-wagmi/src/components/Breadcrumbs.tsx:1]: small fixes in this file to avoid having console errors

File: examples/web-wagmi/src/discovery/UseFeed.tsx
Comment (desfero) [inline examples/web-wagmi/src/discovery/UseFeed.tsx:45]: This change would allow to create a new post when the user has an empty feed.

File: packages/react/src/transactions/adapters/PublicationCallGateway.ts
Comment (desfero) [inline packages/react/src/transactions/adapters/PublicationCallGateway.ts:343]: Should we model some known error when the upload failed? 
Given that `upload` is externally provided by the user it may throw some exceptions that would be good to forward back to the `useCreatePost/Comment` hook in place of `error` so the user may decide to handle those in a way we can't foresee.
Any thoughts?

File: packages/react/src/transactions/adapters/PublicationCallGateway.ts
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/PublicationCallGateway.ts:293]: I know in the original codebase I modelled all these as methods of the same class... I think I violated Open/Closed Principle in doing so and the code seems complicated. Is it thinkable to create an abstract class that a Post Gateway and a Comment Gateway extends by filling the gaps? Will it be easier to manage? Also "create mirror" beyond creating a Lens Publication does have little overlap with the settings of Post and Comment.

Thoughts?

File: packages/react/src/transactions/adapters/PublicationCallGateway.ts
Comment (cesarenaldi) [reply packages/react/src/transactions/adapters/PublicationCallGateway.ts:343]: I've done some work here: https://github.com/lens-protocol/lens-sdk/pull/61

Worth taking into account my other comment: https://github.com/lens-protocol/lens-sdk/pull/57#discussion_r1060600649

There is an opportunity to reduce the complexity of this code. Maybe with a generic or an abstract class?

File: packages/react/src/transactions/adapters/__tests__/PublicationCallGateway.spec.ts
Comment (krzysu) [inline packages/react/src/transactions/adapters/__tests__/PublicationCallGateway.spec.ts:112]: these two mock functions, is it fine to keep them here or they should be places in `@lens-protocol/api-bindings/mocks` as all other mocks?

**Review by krzysu — APPROVED** [review]
I wasn't able to test it in web-wagmi app as I don't have Polygon Matic in my test wallet, but the rest looks good 👍

File: packages/react/src/transactions/adapters/PublicationCallGateway.ts
Comment (desfero) [reply packages/react/src/transactions/adapters/PublicationCallGateway.ts]: Seems there were no use cases for an abstract class as post/comments are both following separate code paths. 
They share some common utils (which are already separate functions) and upload handler logic (which for now is very simple, once #58 is merged we would need to create a new `UploadHandler` class that handles catching and rehrowing upload errors) so I refactored them as 2 separate classes.

**Comment by cesarenaldi** [comment]
> I wasn't able to test it in web-wagmi app as I don't have Polygon Matic in my test wallet, but the rest looks good 👍

@krzysu in general, did you try the instructions in the alert? it suggests how to get Mumbai MATIC...
then the example upload function should guide you to fund your Bunldr wallet to upload to Aerwave.

File: packages/react/src/transactions/adapters/PublicationCallGateway.ts
Comment (cesarenaldi) [reply packages/react/src/transactions/adapters/PublicationCallGateway.ts]: Composition over inheritance feels a better choice, I totally agree.

**Comment by krzysu** [comment]
> > I wasn't able to test it in web-wagmi app as I don't have Polygon Matic in my test wallet, but the rest looks good 👍
> 
> @krzysu in general, did you try the instructions in the alert? it suggests how to get Mumbai MATIC... then the example upload function should guide you to fund your Bunldr wallet to upload to Aerwave.

oh, I was sure I needed Polygon MATIC to upload to aerwave. I have Mumbai MATIC and I've just checked that creating comments works from the example web-wagmi app. 👍

File: packages/react/src/transactions/adapters/useCreateCommentController.ts
Comment (krzysu) [inline packages/react/src/transactions/adapters/useCreateCommentController.ts:52]: maybe we should await here?
When using the example web-wagmi app, the text input is cleared even before signing the first message in the wallet. If the user rejects the message, the comment is lost.

File: examples/web-wagmi/src/publications/UseCreateComment.tsx
Comment (krzysu) [inline examples/web-wagmi/src/publications/UseCreateComment.tsx:44]: can we show somewhere a notification/toast/message in the example component when the comment was created successfully?

File: packages/react/src/transactions/adapters/useCreateCommentController.ts
Comment (cesarenaldi) [reply packages/react/src/transactions/adapters/useCreateCommentController.ts:52]: I agree. When @desfero pointed out the issue with upload publication metadata errors and I looked into it, I ended up awaiting for this promise so to let possible exceptions during the upload bubble up to the `useCreatePost` hook. I think it should happen the same here and in `useCreateComment`.

File: packages/react/src/transactions/adapters/useCreateCommentController.ts
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/useCreateCommentController.ts:29]: Totally agree in deferring the creation of gateway now that is no longer a shared resource.

File: examples/web-wagmi/src/publications/UseCreateComment.tsx
Comment (cesarenaldi) [reply examples/web-wagmi/src/publications/UseCreateComment.tsx:44]: This is an interesting topic. I created a separate ticket to discuss how to move this proposal: https://avara.height.app/lens-sdk/T-2943

File: packages/react/src/transactions/adapters/__tests__/PublicationCallGateway.spec.ts
Comment (desfero) [reply packages/react/src/transactions/adapters/__tests__/PublicationCallGateway.spec.ts]: moved

File: packages/react/src/transactions/adapters/useCreateCommentController.ts
Comment (desfero) [reply packages/react/src/transactions/adapters/useCreateCommentController.ts]: Right, was waiting for #61 to be merged to rebase and fix also catching the `uploadHandler` error.

**Comment by desfero** [comment]
@cesarenaldi @krzysu should be ready for review. 
I'm working on a follow-up PR that would add a `PendingComment` type to the `publications` query so we can have a similar UI that the `useCreatePost` example has where the post dynamically appears after posting.

**Review by krzysu — APPROVED** [review]
LGTM!

**Review by cesarenaldi — APPROVED** [review]

**Pull request merged**
